src/bindings.cpp

```diff
@@ -86,26 +134,59 @@ extern "C"
         return player.get()->isInited() ? 1 : 0;
     }
 
-    /// Load a new sound to be played once or multiple times later
+    /// Load a new sound to be played once or multiple times later.
     ///
-    /// [completeFileName] the complete file path
+    /// After loading the file, the [fileLoadedCallback] will call the 
+    /// Dart function defined with [setDartEventCallback] which gives back
+    /// the error and the new hash.
+    ///
+    /// [completeFileName] the complete file path.
     /// [loadIntoMem] if true Soloud::wav will be used which loads
     /// all audio data into memory. This will be useful when
     /// the audio is short, ie for game sounds, mainly used to prevent
     /// gaps or lags when starting a sound (less CPU, more memory allocated).
     /// If false, Soloud::wavStream will be used and the audio data is loaded
     /// from the given file when needed (more CPU, less memory allocated).
     /// See the [seek] note problem when using [loadIntoMem] = false
-    /// [hash] return hash of the sound
+    /// [hash] return the hash of the sound
     /// Returns [PlayerErrors.noError] if success
-    FFI_PLUGIN_EXPORT enum PlayerErrors loadFile(
+    FFI_PLUGIN_EXPORT void loadFile(
         char *completeFileName,
-        bool loadIntoMem,
+        bool loadIntoMem)
+    {
+        // this check is already been done in Dart
+        if (!player.get()->isInited())
+            return;
+
+        Player *p = player.get();
+        unsigned int hash = 0;
+        // std::thread loadThread([p, completeFileName, loadIntoMem, hash]()
+        //                        {
+        PlayerErrors error = p->loadFile(completeFileName, loadIntoMem, &hash);
+        // printf("*** LOAD FILE FROM THREAD error: %d  hash: %u\n", error,  *hash);
```

Comment:
Are the commented out lines here still needed?

Reply:
I would leave to remember myself to try detaching a loading file thread natively.

Reply:
Fair enough. Let's keep them.